service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchAwaiter.kt

```diff
@@ -88,6 +101,14 @@ class BatchAwaiter(
               dbRunPartition.backfilled_matching_record_count += initialBatch.matching_record_count
               dbRunPartition.scanned_records_per_minute = scannedRateCounter.projectedRate()
               dbRunPartition.matching_records_per_minute = matchingRateCounter.projectedRate()
+              if (dbRunPartition.precomputing_done && dbRunPartition.matching_records_per_minute!! > 0) {
+                val remaining = (dbRunPartition.computed_matching_record_count
+                    - dbRunPartition.backfilled_matching_record_count)
+                val etaMillis = remaining / (dbRunPartition.matching_records_per_minute!! / 60) * 1000
```

Comment:
Does this blow up when matching_records_per_minute is small or 0?

Reply:
I see the 0 check above now but if that number < 30 I think it will still blow up?

Reply:
yeah good call

Reply:
fixed by calculating eta minutes first (as a double to not lose precision)